### DISCUSSION TITLE
`register_uri_scheme_protocol` behaves differently on different platforms! How to get a real custom uri scheme on Windows?

### DISCUSSION THREAD

**Discussion by HuakunShen** [tauri-apps/tauri#10868] [Q&A]
I wrote a file server with `.register_uri_scheme_protocol` on mac to serve extensions (static websites) and load with iframe.

For example, 
- `ext://ext-a/`
- `ext://ext-a/about.html`

I could parse the "domain" and load from different folders without worrying about setting baseUrl.
The domain section tells me `ext-a` is the folder to load from.

It worked perfectly, but when I move to windows, `ext://` is not recognized. `convertFileSrc("", "ext")` returns `http://ext.localhost/`.
On windows I can't have a domain. Thus the url becomes `http://ext.localhost/ext-a/about.html`.

With this, baseUrl have to be configured for frontend project and could result in lots of trouble.

Is it possible to have a real custom uri scheme on Windows?

**Comment by HuakunShen** [comment]
I just saw that the deep link plugin now supports all platforms. 
Not sure if deep link could solve the problem.

**Reply by FabianLars** [reply]
No, deep links do not help you here.

**Reply by HuakunShen** [reply]
Is there sample code for deep-link? The documentation doesn't have much info.
I called `isRegistered` on mac but got error `Unhandled Promise Rejection: unsupported platform`

**Comment by FabianLars** [comment]
> Is it possible to have a real custom uri scheme on Windows?

Not with built-in Tauri settings, see https://github.com/tauri-apps/wry/issues/1129 but you could use https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2customschemeregistration?view=webview2-dotnet-1.0.2106-prerelease via https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.with_webview (not sure about that, it probably has to be done before the webview is fully created)

**Reply by HuakunShen** [reply]
I saw your comments saying that this API is recently added. Is there plan for tauri to implement this url scheme?
I don't think I have the knowledge to implement this in Rust.

**Reply by FabianLars** [reply]
> Is there plan for tauri to implement this url scheme?

As i said in the linked issue, right now not until we drop support for windows 7&8 which won't happen in v2.
